syntax = "proto3";

package raft;

message Log {
	int32 term = 1;
	int32 index = 2;
	string command = 3;
}

message VoteArgs {
	int32 term = 1;
	int32 candidateId = 2; 
	int32 lastLogIndex = 3;
	int32 lastLogTerm = 4;
}

message VoteReply {
	int32 term = 1;
	bool voteGranted = 2;
}

message AppendEntriesArgs {
	int32 term = 1;
	int32 leaderId = 2;
	int32 prevLogIndex = 3;
	int32 prevLogTerm = 4;
	repeated Log entries = 5;
	int32 leaderCommit = 6;
}

message AppendEntriesReply {
	int32 term = 1;
	bool success = 2;
}
